src/win/fs.c

```diff
@@ -1563,6 +1576,29 @@ static void fs__copyfile(uv_fs_t* req) {
 
   overwrite = flags & UV_FS_COPYFILE_EXCL;
 
+  error = fs__stat_impl_from_path(req->fs.info.new_pathw, 0, &new_statbuf);
+  if (error != 0 && error != ERROR_FILE_NOT_FOUND) {
+    SET_REQ_WIN32_ERROR(req, error);
+    return;
+  }
+
+  /* if dst file exists, fstat src file and check if both files are the same,
+   * otherwise CopyFileW directly
+   */
+  if (error == 0) {
+    error = fs__stat_impl_from_path(req->file.pathw, 0, &statbuf);
+    if (error != 0) {
+      SET_REQ_WIN32_ERROR(req, error);
+      return;
+    }
+
+    if (statbuf.st_dev == new_statbuf.st_dev &&
```

Comment:
Is it possible to just call `CopyFileW()`, and only check if the source and destination are the same if it fails with `EBUSY`?

Reply:
I don't really know. Do we know for sure that `ERROR_LOCK_VIOLATION` or `ERROR_SHARING_VIOLATION` are only returned when both files are the same, or are there other cases? I assumed the later.

Reply:
If we aren't sure on that point, couldn't we just check if they are the same on any error? I'm just trying to avoid extra work in the common case if possible.

Reply:
Sorry, I had misread your initial comment. Yes, I think it's a good idea to make the check only if it fails with `EBUSY`.